# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/BytecodeUtil.h
# Commit: d9c1e44eb0ad
# Full Hash: d9c1e44eb0addfc3e44ae299c86381128aa08e7f
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-15 21:49:48
# Regressor Bug: 1601599
# File Overlap Count: 1
# Description:
#   Bug 1601599 part 4 - Support Ion OSR at all loops. r=tcampbell
#   
#   Depends on D56700
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56701
# ==============================================================================

diff -r 93842c3971f8 -r d9c1e44eb0ad js/src/vm/BytecodeUtil.h
--- a/js/src/vm/BytecodeUtil.h	Sun Dec 15 11:33:36 2019 +0000
+++ b/js/src/vm/BytecodeUtil.h	Sun Dec 15 11:34:34 2019 +0000
@@ -254,19 +254,12 @@
 
 static inline unsigned LoopHeadDepthHint(jsbytecode* pc) {
   MOZ_ASSERT(*pc == JSOP_LOOPHEAD);
-  return GET_UINT8(pc + 4) & 0x7f;
+  return GET_UINT8(pc + 4);
 }
 
-static inline bool LoopHeadCanIonOsr(jsbytecode* pc) {
+static inline void SetLoopHeadDepthHint(jsbytecode* pc, unsigned loopDepth) {
   MOZ_ASSERT(*pc == JSOP_LOOPHEAD);
-  return GET_UINT8(pc + 4) & 0x80;
-}
-
-static inline void SetLoopHeadDepthHintAndFlags(jsbytecode* pc,
-                                                unsigned loopDepth,
-                                                bool canIonOsr) {
-  MOZ_ASSERT(*pc == JSOP_LOOPHEAD);
-  uint8_t data = std::min(loopDepth, unsigned(0x7f)) | (canIonOsr ? 0x80 : 0);
+  uint8_t data = std::min(loopDepth, unsigned(UINT8_MAX));
   SET_UINT8(pc + 4, data);
 }
 